### DISCUSSION TITLE
Reading View prototype

### DISCUSSION THREAD

**Discussion by NickEastNL** [chuangcaleb/obsidian-fountain-editor#33] [Ideas]
I've added rudimentary support for Reading View. It isn't fully functional yet (sections, comments, and the like aren't formatted yet as I don't use them), though it shouldn't be hard to add it (since most of that shouldn't be rendered in a "read view" anyway). I also had to make a compromise to format dialogue with multiple paragraphs. For that reason, I haven't set up a Pull Request.

Due to a limitation in Obsidian's API, a Markdown "Post" Processor can only work with the parsed HTML, rather than the Markdown source. Since whitespace is removed, the double spaces to mark a paragraph break in dialogue cannot be used to format the HTML. A workaround I applied is to put a `\` (backslash) character on the empty line. As this is an escape character, Obsidian will not render it, but the parsed Markdown renders two line breaks (`<br>`) instead of one. Since it's also an actual character, Fountain will parse both it and the following line as dialogue. Then as I iterate through each line of dialogue to create paragraphs, I simply include the empty "paragraph" to simulate the paragraph break (as dialogue doesn't have top/bottom margins). In the case of exporting the Fountain content, it shouldn't be too hard to replace `\` with two spaces to conform to the spec.

Everything else should work fine by simply applying the Fountain rules to the text content of each paragraph. It produces a bit of weird HTML (since each paragraph is nested in a DIV, and since lines starting with `>` produce block quotes), but the rendering should actually be correct. I've added the relevant CSS.

Another reason I didn't provide a Pull Request is that I switched to using Courier Prime as the font instead of Courier New, since that's a modern and recommended screenplay font. I didn't have a lot of time to set all this up, since I'm actively using Fountain at the moment and needed to get Reading View to work ASAP, so things might not be very clean and efficient.

I'm providing the diff here to see what I added/changed: https://github.com/chuangcaleb/obsidian-fountain-editor/compare/master...NickEastNL:obsidian-fountain-editor:read-view

![example](https://github.com/chuangcaleb/obsidian-fountain-editor/assets/8384282/977e8f83-b8a4-4787-930f-83e3f1b8e0e2)

**Comment by chuangcaleb** [comment]
YO this is NEAT thanks @NickEastNL! And it's cool that you also walked through your thought process.

We could modify the Fountain parsing rules to ignore lines starting with `\` — while not official spec, it's an option.

Yeah HTML rendering is weird, each line is its own `<div>` right. 

On the note of fonts, yeah we can just change [the source CSS](https://github.com/chuangcaleb/obsidian-live-fountain/blob/daba3fb7b3d0603ca21d175292edaabe61c0ddb7/src/styles/base.css#L30-L31) to Courier Prime, why not, shall we?

Well please feel free to open a PR when you're ready!

**Reply by NickEastNL** [reply]
Hey, thanks for the quick response!

Before I set up a PR, I intend to put the plugin through its paces with my workflow and do some thorough testing. I also want to make sure it's more-or-less "ready" rather than half-baked and half-finished, cleaning up some code and trying to see if I can improve my workaround.

**Reply by Sanabria** [reply]
@NickEastNL Thank you for expanding the plugin to include support for reading mode. Could you provide an estimated timeline for when this feature will be publicly available?

**Reply by NickEastNL** [reply]
> @NickEastNL Thank you for expanding the plugin to include support for reading mode. Could you provide an estimated timeline for when this feature will be publicly available?

Shortly after getting to this point, I had to pause the work I was doing with this plugin, so I haven't had a chance to improve it to make it suitable for a PR. I'm getting back to this very soon, and it shouldn't require too much extra work. However, I'm doing this based on my specific use case, so it's not very complete, and depends on approval. As such, I can't promise an ETA.
